sdk/appcenter-auth/src/main/java/com/microsoft/appcenter/auth/Auth.java

```diff
@@ -461,24 +469,58 @@ private synchronized void initAuthenticationClient(String configurationPayload)
             JSONObject configuration = new JSONObject(configurationPayload);
             String identityScope = configuration.getString(IDENTITY_SCOPE);
             String authorityUrl = null;
+            String type = null;
+            String audienceType = null;
             JSONArray authorities = configuration.getJSONArray(AUTHORITIES);
             for (int i = 0; i < authorities.length(); i++) {
                 JSONObject authority = authorities.getJSONObject(i);
                 if (authority.optBoolean(AUTHORITY_DEFAULT) && AUTHORITY_TYPE_B2C.equals(authority.getString(AUTHORITY_TYPE))) {
-                    authorityUrl = authority.getString(AUTHORITY_URL);
-                    break;
+                    type = AUTHORITY_TYPE_B2C;
+                } else if (authority.optBoolean(AUTHORITY_DEFAULT) && AUTHORITY_TYPE_AAD.equals(authority.getString(AUTHORITY_TYPE))) {
+                    type = AUTHORITY_TYPE_AAD;
                 }
+
+                JSONObject audience = authority.optJSONObject(Audience);
+                if (audience != null) {
+                    audienceType = audience.getString(Audience_Type);
+                }
+
+                authorityUrl = authority.optString(AUTHORITY_URL);
             }
-            if (authorityUrl != null) {
 
-                /* The remaining validation is done by the library. */
-                mAuthenticationClient = new PublicClientApplication(mContext, getConfigFile());
-                mAuthorityUrl = authorityUrl;
-                mIdentityScope = identityScope;
-                AppCenterLog.info(LOG_TAG, "Auth service configured successfully.");
+            if (type.equals(AUTHORITY_TYPE_B2C)) {
+                if (authorityUrl == null) {
+                    throw new IllegalStateException("B2C authority is configured incorrectly. Authority url is mandatory.");
```

Comment:
I would argue checks like this are overzealous. I'm assuming MSAL already throws relevant errors in these cases.

Clement mentioned we need 100% code coverage, which if I understand correctly means that each of these failure lines will require a separate unit test to test it, which feels like a lot of code, and all these checks and all those unit tests seem like more code than they are worth for what it's doing.

Reply:
Yes the big benefit of coverage forces us to use less if and only ones that actually matter and avoid dead code :)

Reply:
I added unit tests for this change.